playlist_souffle/delivery/aws_lambda/souffle_playlist.py

```diff
@@ -52,17 +55,17 @@ def main(event, context):
 
     # Setup spotipy client
     try:
-        spotify = Spotify(access_token)
-    except SpotifyException as e:
+        spotify = SpotifyGateway(access_token)
+    except SouffleSpotifyError as e:
         return generate_spotify_exception_response(e)
 
 
     # Souffle playlist
     try:
-        souffled_playlist_uri = souffle_playlist(playlist_uri, shuffle_by, user_id, spotify)
+        souffled_playlist_uri = souffle_playlist(spotify, playlist_uri, user_id, shuffle_by)
```

Comment:
cool! also, I think I'd do: spotify, user_id, playlist_uri, shuffle_by.

not exactly sure how strong my logic is. But services, than the most likely thing to repeat in multiple requests (for example, you could have the same user), then the shuffle_by because it's an option describing the output, rather than one of the inputs.

what do you think?